### PR TITLE
Book mappings

### PR DESCRIPTION
**Pull request by emilyploszaj**
`BookEditScreen` and some associated classes were kinda a mess and missing a lot of stuff, so I mapped them. A couple mappings were straight up wrong as well, but it's understandable given how much time I had to spent figuring out what everything did, so I fixed those too.

### CODE REVIEW

**Review by jaskarth — APPROVED** [review]

**Comment by enbrain** [comment]
FYI: There is a PR that maps book stuff https://github.com/FabricMC/yarn/pull/1941

File: mappings/net/minecraft/client/util/SelectionManager.mapping [hunk 1]
```diff
@@ -6,38 +6,63 @@ CLASS net/minecraft/class_3728 net/minecraft/client/util/SelectionManager
...
 	METHOD method_16199 insert (C)Z
 		ARG 1 c
 	METHOD method_16200 getSelectedText (Ljava/lang/String;)Ljava/lang/String;
+		ARG 1 string
 	METHOD method_16201 getSelectionStart ()I
 	METHOD method_16202 handleSpecialKey (I)Z
 		ARG 1 keyCode
 	METHOD method_16203 getSelectionEnd ()I
-	METHOD method_16204 moveCaretToEnd ()V
+	METHOD method_16204 moveCursorToEnd ()V
```
Comment (liach) [inline mappings/net/minecraft/client/util/SelectionManager.mapping]: How does this one relate to the version that take a boolean?

File: mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping [hunk 2]
```diff
@@ -49,11 +49,43 @@ CLASS net/minecraft/class_473 net/minecraft/client/gui/screen/ingame/BookEditScr
...
+		ARG 1 lines
+	METHOD method_27581 drawCursor (Lnet/minecraft/class_4587;Lnet/minecraft/class_473$class_5234;Z)V
 		ARG 1 matrices
+		ARG 2 position
+		ARG 3 atEnd
+	METHOD method_27582 screenPositionToAbsolutePosition (Lnet/minecraft/class_473$class_5234;)Lnet/minecraft/class_473$class_5234;
+		ARG 1 position
+	METHOD method_27583 getRectFromCorners (Lnet/minecraft/class_473$class_5234;Lnet/minecraft/class_473$class_5234;)Lnet/minecraft/class_768;
+		ARG 1 position1
+		ARG 2 position2
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/ingame/BookEditScreen.mapping]: Maybe `start` `end`?

**Review by liach — COMMENTED** [review]
Fyi you can just use `pos` (allowed by convention) if `position` is too long.

**Review by YanisBft — COMMENTED** [review]
Why is this targeting 1.16.5?

**Comment by liach** [comment]
Imo targetting 1.16.5 isn't too problematic. Does anyone know if books changed a lot during the snapshots?

About #1941, I think this one is more comprehensive.

**Comment by YanisBft** [comment]
No there is no change at all, could be merged in the two branches easily (it's better to target latest and then backport than the reverse tho)

**Comment by liach** [comment]
Since we are already reviewing here, I guess we can just cherry pick the commit from release to snapshot. Not worth to move the discussion around to a new pr

**Comment by emilyploszaj** [comment]
Oh sorry, I was under the impression that people target stable versions if they can, I'll target snapshots in the future

File: mappings/net/minecraft/client/util/SelectionManager.mapping (hunk 1 above)
Comment (emilyploszaj) [reply mappings/net/minecraft/client/util/SelectionManager.mapping]: Turns out the method was improperly named, it actually moves the cursor to the start, fixed

**Review by YanisBft — APPROVED** [review]

**Review by liach — APPROVED** [review]

**Comment by liach** [comment]
A side note: This is on 1.16.5 (which is perfectly fine as this maps missing stuff for 1.16.5). Need a cherry pick onto 21w11a after this is merged.

**Comment by modmuss50** [comment]
Needs to go into the snapshot first, please.

**Comment by liach** [comment]
feel free to keep this one open and still target 1.16.5. it can get merged after it's merged in snapshot

**Comment by IotaBread** [comment]
Is this going to be merged?

**Comment by liach** [comment]
I think so if this is reopened and targets 1.16.5 branch

**Comment by IotaBread** [comment]
Can I open a PR cherry-picking this if it isn't reopened after a while?

**Comment by liach** [comment]
You can do one right now imo.

**Pull request closed without merging**